web/models/class/class_helper.ex

```diff
@@ -38,6 +38,46 @@ defmodule CoursePlanner.ClassHelper do
   end
 
   defp non_deleted_query do
-    from c in Class , where: is_nil(c.deleted_at)
+    from c in Class,
+      preload: [{:offered_course, :course}],
+      where: is_nil(c.deleted_at),
+      order_by: [desc: :date, desc: :starting_at]
+  end
+
+  defp get_class_students(offered_course_id) do
```

Comment:
what do you think of adding `has_many :students, through: [:offered_course, :students]` in the class schema?

this way you don't need to add those joins every time, only the preload